cmd/promtool/unittest_test.go

```diff
@@ -125,15 +126,54 @@ func TestRulesUnitTest(t *testing.T) {
 			want: 0,
 		},
 	}
+	reuseFiles := []string{}
+	reuseCount := [2]int{}
 	for _, tt := range tests {
+		if (tt.queryOpts == promqltest.LazyLoaderOpts{}) {
+			reuseFiles = append(reuseFiles, tt.args.files...)
+		}
 		t.Run(tt.name, func(t *testing.T) {
 			if got := RulesUnitTest(tt.queryOpts, nil, false, tt.args.files...); got != tt.want {
 				t.Errorf("RulesUnitTest() = %v, want %v", got, tt.want)
 			}
 		})
 	}
-}
 
+	reTop := regexp.MustCompile(`(?s)^<testsuites\W.*</testsuites>$`)
```

Comment:
nit: this is duplicated from  junitxml_test.go , couldn't we just use encoding/xml to parse the output?